lib/static/new-ui/features/suites/components/SuitesTreeView/utils.ts

```diff
@@ -0,0 +1,389 @@
+                        return [0];
+                    }
+
+                    return childrenWeight;
+                } else if (currentSortExpression.type === SortType.ByTestsCount) {
+                    if (!childrenWeight) {
+                        notifyOfUnsuccessfulWeightComputation();
+                        return [0];
+                    }
+
+                    return childrenWeight;
+                }
+                break;
+            }
+            case EntityType.Browser: {
+                if (currentSortExpression.type === SortType.ByName) {
+                    return [treeNode.data.title.join(' ')];
+                } else if (currentSortExpression.type === SortType.ByFailedRetries) {
+                    const browser = browsers[treeNode.data.entityId];
+                    const groupId = getGroupId(treeNode.data);
+
+                    return [browser.resultIds.filter(resultId =>
+                        (isFailStatus(results[resultId].status) || isErrorStatus(results[resultId].status)) &&
+                        (!groupId || groups[groupId].resultIds.includes(resultId))
+                    ).length];
+                } else if (currentSortExpression.type === SortType.ByTestsCount) {
+                    const browser = browsers[treeNode.data.entityId];
+                    const groupId = getGroupId(treeNode.data);
+                    const retriesCount = groupId ? browser.resultIds.filter(resultId => groups[groupId].resultIds.includes(resultId)).length : browser.resultIds.length;
+
+                    return [1, retriesCount];
+                }
+                break;
+            }
+        }
+
+        notifyOfUnsuccessfulWeightComputation();
+        return [0];
+    };
+
+    const aggregateWeights = (weights: TreeNodeWeight[]): TreeNodeWeight => {
+        if (!currentSortExpression || currentSortExpression.type === SortType.ByName) {
+            return [0];
+        }
+
+        if (currentSortExpression.type === SortType.ByFailedRetries || currentSortExpression.type === SortType.ByTestsCount) {
+            return weights.reduce((acc, weight) => {
+                const newAcc = acc.slice(0);
+                for (let i = 0; i < weight.length; i++) {
+                    newAcc[i] = (acc[i] ?? 0) + weight[i];
+                }
+                return newAcc;
+            }, new Array(weights[0]?.length));
+        }
+
+        return [0];
+    };
+
+    // Recursive tree sort. At each level of the tree, it does the following:
+    // 1. Compute weights of the current branch
+    // 2. Sort current level according to weights
+    // 3. Return sorted current level and aggregate of all weights at current level
+    const sortAndGetWeight = (treeNodes: TreeNode[]): TreeWeightedSortResult => {
+        const treeNodesCopy = treeNodes.slice(0);
+        const weights: Record<string, TreeNodeWeight> = {};
+
+        treeNodesCopy.forEach((treeNode, index) => {
```

Comment:
hard core =)

Reply:
Agree! So I decided to fix it. Replied here: https://github.com/gemini-testing/html-reporter/pull/621#discussion_r1875054827